app/domain/authentication/authn_k8s/application_identity.rb

```diff
@@ -141,7 +141,7 @@ def prefixed_k8s_annotations prefix
           annotation_name = a.values[:name]
 
           annotation_name.start_with?(prefix) &&
-            # Verify we take only annotations from the same level
+            # verify we take only annotations from the same level
```

Comment:
I'd add a code example to the comments here showing an actual name and prefix so the reader can see what this logic is doing.  eg, why does "same level" mean that the number of parts of one is one more than the number of parts of the other?

also consider breaking into multiple statements for clarity.